File: backend/src/main/java/backend/com/edtech/question/service/QuestionService.java
package backend.com.edtech.question.service;

import backend.com.edtech.question.dto.QuestionRequestDto;
import backend.com.edtech.question.entity.Question;

import java.util.List;

public interface QuestionService {

    Question createQuestion(QuestionRequestDto dto);

    List<Question> getQuestionsByExam(Long examId);

    Question updateQuestion(Long id, QuestionRequestDto dto);

    void deleteQuestion(Long id);

}

File: edTech-frontend/src/modules/question/services/questionService.ts
import axios from "axios";

const BASE_URL = "http://localhost:8080/api/questions";

export const createQuestion = (question: any) => {
    return axios.post(BASE_URL, question);
};

export const getQuestionsByExam = (examId: number) => {
    return axios.get(`${BASE_URL}/exam/${examId}`);
};

export const updateQuestion = (
    id: number,
    question: any
) => {
    return axios.put(`${BASE_URL}/${id}`, question);
};

export const deleteQuestion = (id: number) => {
    return axios.delete(`${BASE_URL}/${id}`);
};